### ISSUE TITLE
Numerical issues in multiclass_logistic_loss

### ISSUE THREAD

**Issue by jvlmdr** [google/jaxopt#243]
The current definition of multiclass_logistic_loss has a minor numerical issue:
```python
>>> jaxopt.loss.multiclass_logistic_loss(0, jnp.array([1e9, 1e9]))
DeviceArray(0., dtype=float32)  # expect log(2)
```

It is also not possible to use `-inf` to mask unused logits:
```python
>>> jaxopt.loss.multiclass_logistic_loss(0, jnp.array([0, 0, -jnp.inf]))
DeviceArray(nan, dtype=float32)  # expect log(2)
```

And, while less critical, it would be nice if it could handle positive infinity:
```python
>>> jaxopt.loss.multiclass_logistic_loss(0, jnp.array([jnp.inf, 0]))
DeviceArray(nan, dtype=float32)  # expect 0
```

One alternative implementation is:
```python
def multiclass_logistic_loss(label: int, logits: jnp.ndarray) -> float:
    return logsumexp((logits - logits[label]).at[label].set(0.))
```

**Comment by mblondel** [comment]
Thanks a lot for reporting! We would welcome a fix and the corresponding unit tests.

**Comment by mblondel** [comment]
Fixed in #245.
